x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/GetInferenceFieldsActionRequestTests.java

```diff
@@ -0,0 +1,122 @@
+                instance.getFields(),
+                instance.resolveWildcards(),
+                instance.useDefaultFields(),
+                instance.getQuery(),
+                instance.getIndicesOptions()
+            );
+            case 1 -> new GetInferenceFieldsAction.Request(
+                instance.getIndices(),
+                randomValueOtherThan(instance.getFields(), GetInferenceFieldsActionRequestTests::randomFields),
+                instance.resolveWildcards(),
+                instance.useDefaultFields(),
+                instance.getQuery(),
+                instance.getIndicesOptions()
+            );
+            case 2 -> new GetInferenceFieldsAction.Request(
+                instance.getIndices(),
+                instance.getFields(),
+                randomValueOtherThan(instance.resolveWildcards(), ESTestCase::randomBoolean),
+                instance.useDefaultFields(),
+                instance.getQuery(),
+                instance.getIndicesOptions()
+            );
+            case 3 -> new GetInferenceFieldsAction.Request(
+                instance.getIndices(),
+                instance.getFields(),
+                instance.resolveWildcards(),
+                randomValueOtherThan(instance.useDefaultFields(), ESTestCase::randomBoolean),
+                instance.getQuery(),
+                instance.getIndicesOptions()
+            );
+            case 4 -> new GetInferenceFieldsAction.Request(
+                instance.getIndices(),
+                instance.getFields(),
+                instance.resolveWildcards(),
+                instance.useDefaultFields(),
+                randomValueOtherThan(instance.getQuery(), GetInferenceFieldsActionRequestTests::randomQuery),
+                instance.getIndicesOptions()
+            );
+            case 5 -> new GetInferenceFieldsAction.Request(
+                instance.getIndices(),
+                instance.getFields(),
+                instance.resolveWildcards(),
+                instance.useDefaultFields(),
+                instance.getQuery(),
+                randomValueOtherThan(instance.getIndicesOptions(), GetInferenceFieldsActionRequestTests::randomIndicesOptions)
+            );
+            default -> throw new AssertionError("Invalid value");
+        };
+    }
+
+    private static Set<String> randomIndentifierSet() {
+        return randomSet(0, 5, ESTestCase::randomIdentifier);
+    }
+
+    private static Map<String, Float> randomFields() {
+        return randomMap(0, 5, () -> Tuple.tuple(randomIdentifier(), randomFloat()));
+    }
+
+    private static String randomQuery() {
+        return randomBoolean() ? randomAlphaOfLength(10) : null;
+    }
+
+    private static IndicesOptions randomIndicesOptions() {
+        // This isn't an exhaustive list of possible indices options, but there are enough for effective serialization tests
+        return switch (between(0, 9)) {
```

Comment:
Why do a switch here, rather than just `randomBoolean`s to construct a new `IndicesOptions.fromOptions`?

Reply:
Oh, that's subtle and crappy. Thanks for the explanation. May be worth a comment here.